bl/validation/k8sValidator.go

```
@@ -106,9 +106,13 @@ func (val *K8sValidator) validateResource(filepath string) (bool, []error, error
 	if err != nil {
 		return false, []error{}, fmt.Errorf("failed opening %s: %s", filepath, &InvalidK8sSchemaError{ErrorMessage: err.Error()})
 	}
+	defer func() {
+		_ = f.Close()
```

There's no need to wrap the defer inside a function, simply use `defer f.Close()`. In addition, we don't want to ignore errors so if for any reason closing the file failed we would want to return the error.

---

As of my understanding the way I defined the defer function is behaving exactly the same as
`defer f.Close()` which is ignoring return values.
If there is need of returning the error it has to be done explicitly, let me know if I should refactor the code accordingly.

---

Hey @tipok! Haha, you are absolutely correct, I completely missed it, sorry for the trouble! We decided we want to ignore the error, so simply use `defer f.Close()`

---

ok, I'll change it, thank you. Just my 2 cents on that topic: 
1. Some linter would show issues on that because errors are not handled correctly 
2. As discovered in that thread it's not that obvious what happens with returned values on defer if not handled explicitly (as in my solution)

 Therefore I personally do prefer handling it explicitly so it's totally clear what is happening.

---

Thank you! I understand what you mean. However, we prefer to continue with this until we'll gain more knowledge about how we would want to handle it.